### PR TITLE
Fix up C++ example in README

### LINKED ISSUE #153: Cpp optimization example compilation issue
**Describe the bug**
Seems to be an issue with the bearing factor example referenced in the readme. The sym::Factor<double>::Hessian referenced does not find the correct template to make these code-genned factors work in cpp.

**To Reproduce**
```estimation/test/symforce_test.cc:37:17: error: no matching function for call to 'Hessian'
                sym::Factor<double>::Hessian(&sym::BearingFactor,
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
external/symforce/include/symforce/opt/./internal/../factor.h:257:17: note: candidate template ignored: couldn't infer template argument 'Functor'
  static Factor Hessian(Functor func, const std::vector<Key>& keys_to_func,
                ^
external/symforce/include/symforce/opt/./internal/../factor.h:245:17: note: candidate function template not viable: requires 2 arguments, but 3 were provided
  static Factor Hessian(Functor func, const std::vector<Key>& keys);
```                ^

**Expected behavior**
factor should correctly be added to optimization problem in the example given.

**Environment (please complete the following information):**
 - OS and version: Ubuntu Linux 20.04
 - Python version 3.8.5
 - SymForce Version: aba8ef7

**Additional context**
I am using a different build environment (bazel), but I still can't cooerce these templates to find the correct version of factor / hessian template for the codegenned versions of bearing and odometry factor. The factors are correctly code-genned it seems. I am likely doing something dumb...

### PR DESCRIPTION
**Pull request by aaron-skydio**
Fixes #153

### CODE REVIEW

**Review by hmartiro — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -176,6 +176,12 @@ See the [Epsilon Tutorial](https://symforce.org/notebooks/epsilon_tutorial.html)
 
 We will model this problem as a factor graph and solve it with nonlinear least-squares.
 
+For this example in particular, we need to make sure we're on the `symengine` symbolic backend, and not `sympy`:
+
+```python
+assert symforce.get_backend() == "symengine"
+```
```
Comment (hayk-skydio) [inline README.md:183]: we should be able to kill this after #163 ? probably just worth doing now

File: README.md [hunk 2]
```diff
@@ -482,9 +489,25 @@ sym::Optimizer<double> optimizer(
...
+values.Set({'d', 1}, 1.4);
+values.Set({'a', 0, 0}, 55 * M_PI / 180);
+values.Set({'a', 0, 1}, 245 * M_PI / 180);
+values.Set({'a', 0, 2}, -35 * M_PI / 180);
+values.Set({'a', 1, 0}, 95 * M_PI / 180);
+values.Set({'a', 1, 1}, 220 * M_PI / 180);
+values.Set({'a', 1, 2}, -20 * M_PI / 180);
+values.Set({'a', 2, 0}, 125 * M_PI / 180);
+values.Set({'a', 2, 1}, 220 * M_PI / 180);
+values.Set({'a', 2, 2}, -20 * M_PI / 180);
```
Comment (hayk-skydio) [inline README.md:509]: maybe better to create a 2D array of the values like `{{55, 245, -35}, {95, 220, -20}, ...}` then do a double loop and set the a keys? It should shorten this and get rid of the duplicate M_PI / 180

File: README.md [hunk 3]
```diff
@@ -482,9 +489,25 @@ sym::Optimizer<double> optimizer(
...
+values.Set('e', sym::kDefaultEpsilond);
 
 // Optimize!
```
Comment (hayk-skydio) [inline README.md:512]: I do think it's worth having a fully complete C++ version in examples/robot_2d_triangulation

**Review by hayk-skydio — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (aaron-skydio) [reply README.md]: Killed, that one should merge before or same time as this one

File: README.md (hunk 2 above)
Comment (aaron-skydio) [reply README.md]: Done

File: README.md (hunk 3 above)
Comment (aaron-skydio) [reply README.md:505]: @bradley-solliday-skydio is going to do this

Comment (aaron-skydio) [reply README.md:505]: #167

**Pull request closed without merging**